### PR TITLE
[FIX] merge_records: complete _change_translations_sql method

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
This method was incomplete.

### CODE REVIEW

**Review by HviorForgeFlow — APPROVED** [review]
Works fine! :+1:

**Comment by pedrobaeza** [comment]
Why can't you do things in one query instead of a lot of select?

**Comment by MiquelRForgeFlow** [comment]
What do you mean? I only did a verbatim copy of orm method, but with sql queries instead of using method search.

**Comment by pedrobaeza** [comment]
I mean that you have put 3 SELECT query, and that seems to be a non optimized way to do SQL. Please try to join the final action to do (I understand `DELETE`) in only query.

**Comment by MiquelRForgeFlow** [comment]
Done, fused the queries. But I prefer what it was done before, as it was more readable.

**Comment by pedrobaeza** [comment]
But for logging purposes this is far better, and also for performance. Do you prefer to have 1 logged query or 200 (one per iteration of the loop)?

**Comment by pedrobaeza** [comment]
And this is still not what I was asking for. You still do things in 2 steps. `UPDATE` or `DELETE` queries should include everything, not a previous `SELECT` and then the rest.

**Comment by MiquelRForgeFlow** [comment]
I doubt clients have bbdd with 200 name/lang installed haha but ok, I will improve the way you say.

**Comment by pedrobaeza** [comment]
It was just an example of general problem, but yes, I have customer with 7 languages. Anyway, I think you are not understanding yet the problem. Imagine this simplified case:
`SELECT id FROM res_partner WHERE customer`
`UPDATE res_partner SET name = name WHERE id in %s`
and get the logging with `UPDATE res_partner SET name = name WHERE id in (1, 2, 3, 4....21103)`
instead of a cleaner:
`UPDATE res_partner SET name = name WHERE customer`
The idea is the same for this, but with more difficult statements.

**Comment by MiquelRForgeFlow** [comment]
Done, how it looks now?

File: openupgradelib/openupgrade_merge_records.py [hunk 1]
```diff
@@ -248,26 +248,50 @@ def _change_translations_orm(env, model_name, record_ids, target_record_id,
...
         }, skip_no_result=True)
+    logged_query(
+        env.cr, """
+        DELETE FROM ir_translation it
+        USING (
+            SELECT it.id
+            FROM ir_translation it
+            WHERE it.type = 'model' AND it.res_id IN %(record_ids)s
+                AND it.name like %(model_name)s || ',%%'
+                AND it.res_id != %(target_record_id)s
```
Comment (pedrobaeza) [inline openupgradelib/openupgrade_merge_records.py:287]: Shouldn't this be?
```suggestion
                AND it.res_id IN %(record_ids)s
```

Comment (MiquelRForgeFlow) [reply openupgradelib/openupgrade_merge_records.py:287]: See two lines above, please hehe. This extra line is to discard one of those records

Comment (pedrobaeza) [reply openupgradelib/openupgrade_merge_records.py:287]: Ah, OK, then it's not needed, as `target_record_id` is not included in `record_ids`

Comment (MiquelRForgeFlow) [reply openupgradelib/openupgrade_merge_records.py:287]: Well explained. Updated!

**Pull request merged**